<?xml version="1.0"?>
<launch>

  <arg name="pointing_frame"	default="magnet_tip_link"/>
  <arg name="min_duration"	default="0.01"/>
  <arg name="max_velocity"	default="0.4"/>
  <arg name="target_frame"	default="marker_frame"/>
  <arg name="target_pose"	default="[0.1, 0.10, 0.35, -90, 90, 0]"/>
  <arg name="server"		default="pose_head_tracker"/>
  <arg name="vis"		default="false"/>

  <!-- Spawn tracker client -->
  <node name="pose_head_client"
        pkg="aist_controllers" type="pose_head_client.py"
        output="screen" respawn="false">
    <param name="pointing_frame"	value="$(arg pointing_frame)"/>
    <param name="min_duration"		value="$(arg min_duration)"/>
    <param name="max_velocity"		value="$(arg max_velocity)"/>
    <param name="target_frame"		value="$(arg target_frame)"/>
    <param name="server"		value="$(arg server)"/>
    <rosparam param="target_pose"
	      subst_value="true">$(arg target_pose)</rosparam>
  </node>

  <node name="rqt_plot" pkg="rqt_plot" type="rqt_plot"
	args="/arm_controller/command_debug/points[0]/positions[0]
	      /arm_controller/command_debug/points[0]/positions[1]
	      /arm_controller/command_debug/points[0]/positions[2]
	      /arm_controller/state/actual/positions[0]
	      /arm_controller/state/actual/positions[1]
	      /arm_controller/state/actual/positions[2]"
        output="screen" respawn="false"/>

  <!-- Launch rviz with appropriate launch-prefix and config file. -->
  <node if="$(arg vis)"
	name="$(anon rviz)" pkg="rviz" type="rviz" output="screen"
	args="-d $(find calibration_moveit_config)/launch/moveit.rviz"/>

</launch>
